evap/evaluation/templates/base.html

```diff
@@ -85,49 +83,73 @@
-            // django-dynamic-formsets' "add another" button, so we take a copy of a non-select2'd form here
-            var templateForm = $(".select2tr").last().clone();
+            // These are workarounds around incompatibilities of django, django-dynamic-formsets, and tomselect.
+            // tomselect can't handle already tomselect'd forms that were copies made by
+            // django-dynamic-formsets' "add another" button, so we take a copy of a non-tomselect'd form here
+            var templateForm = $(".tomselecttr").last().clone();
 
             // later we use this class to give this to django-dynamic-formsets
-            // as template and to make sure it does not get select2'd
+            // as template and to make sure it does not get tomselect'd
             templateForm.addClass("form-template").css("display", "none");
             // for some reason django-dynamic-formsets does not hide the checkbox like it should
             templateForm.find(':checkbox').last().attr("type", "hidden");
             // For some reason, django validates this template if it's part of the form, so we insert the copy outside of the form.
-            templateForm.insertAfter($('.select2form'));
+            templateForm.insertAfter($('.tomselectform'));
 
-            // put this into a function for later re-use
-            applySelect2 = function(elem, parent = $(document.body)) {
-                elem.each(function() {
-                    $(this).select2({
-                        language: "{{ LANGUAGE_CODE }}",
-                        placeholder: "{% trans 'Please select...' %}",
-                        minimumResultsForSearch: 15,
-                        minimumInputLength: $(this).children('option').length > 50 ? 3 : 0,
-                        allowClear: this.multiple,
-                        dropdownParent: parent
-                    });
-                });
+            // used in derived templates
+            const tagTomSelectOptions = {
+                create: true,
+                dropdownParent: document.createElement("div"),
             };
-            applySelect2($('select').not('.form-template select').not('[disabled]').not('.no-select2').not('#id_delegate_to'));
-            // handle delegation modal, which includes a dropdown select field
-            applySelect2($('select#id_delegate_to'), $('#delegateSelectionModal'));
 
-            // don't use the placeholder text on disabled select2 elements
-            applySelect2Disabled = function(elem) {
-                elem.select2();
+            class MinimumInputLengthTomSelect extends TomSelect {
+                search(query) {
+                    if(query.length < this.settings.minimumInputLength) {
+                        return this.sifter.prepareSearch(query, {});
+                    }
+                    return super.search(query);
+                }
+            }
+
+            applyTomSelect = function(elements, additionalOptions = {}) {
```

Comment:
(random line of code )

TomSelect has the same issue as we had with select2 in #1638: In all kinds of bad situations, it will make the page scroll to some other location:
* most notably currently: Single select, clearing with backspace
* Also, just like in #1638: When opening the dropdown, scrolling somewhere else, and then closing the dropdown by clicking somewhere, the page will scroll back up.

Reply:
Some of the scrolling is caused by our custom `invalid` listener here:
https://github.com/e-valuation/EvaP/blob/81f1d4a0d06347b9bc4ba546e0933f6e02066228/evap/evaluation/templates/base.html#L136-L139

This can be fixed by just using `scroll-padding-top` instead of this custom `invalid` handler.

The scroll-back-on-focus-lose is caused by tomselect's `dropdown_input` plugin. I opened issue 391 with them.